tests/transforms/test_split_non_commuting.py

```diff
@@ -953,3 +955,93 @@ def test_trainable_hamiltonian_tensorflow(self, grouping_strategy):
         actual = tape.jacobian(cost, params)
 
         assert qml.math.allclose(actual, [-0.5, np.cos(np.pi / 4)])
+
+
+# Single hamiltonian expval measurement to test shot distribution
+ham = qml.Hamiltonian(
+    coeffs=[10, 0.1, 20, 100, 0.2],
+    observables=[
+        qml.X(0) @ qml.Y(1),
+        qml.Z(0) @ qml.Z(2),
+        qml.Y(1),
+        qml.X(1) @ qml.X(2),
+        qml.Z(0) @ qml.Z(1) @ qml.Z(2),
+    ],
+)
+ham_expval = qml.expval(ham)
+total_shots = 1000
+coeffs_per_group = [[10, 20], [0.1, 0.2], [100]]
```

Comment:
Since all these definitions are so short, let's just place them in the individual tests where they are used. Makes it a little easier when debugging the tests if you don't have spend so much time tracking down definitions.